yarn-project/simulator/src/client/simulator.ts

```diff
@@ -85,16 +85,18 @@ export class AcirSimulator {
     const curve = new Grumpkin();
 
     const header = await this.db.getHeader();
+
+    // reserve the first side effect for the tx hash (inserted by the private kernel)
```

Comment:
Plus, won't all side effect counters that come from private eventually be 0?

Reply:
yes, but this is the simulator that runs on the PXE so side effects here are >0.